### PR TITLE
[NEW] Support CAS login + hide login form & password reset according to server

### LINKED ISSUE #1198: [NEW] Support CAS login method

### PR DESCRIPTION
**Pull request by cardoso**
@RocketChat/ios

|Preview|
|-|
||

Closes #1198

### CODE REVIEW

File: Rocket.Chat/Controllers/Auth/CASViewController.swift [hunk 1]
```diff
@@ -0,0 +1,106 @@
+
```
[the remark this answers is not in the document]
Comment (cardoso) [reply Rocket.Chat/Controllers/Auth/CASViewController.swift]: 👎

[the remark this answers is not in the document]
Comment (rafaelks) [reply Rocket.Chat/Controllers/Auth/CASViewController.swift]: Hound is CRAZY these days. Thinking about removing it. :-(

File: Rocket.Chat/Controllers/Auth/AuthViewController.swift [hunk 2]
```diff
@@ -68,7 +82,10 @@ final class AuthViewController: BaseViewController {
            buttonRegister.isHidden = registrationForm != .isPublic
         }
 
-        self.updateAuthenticationMethods()
+        hideViewFields = !(AuthSettingsManager.settings?.isUsernameEmailAuthenticationEnabled ?? true)
+        buttonResetPassword.isHidden = !(AuthSettingsManager.settings?.isPasswordResetEnabled ?? true)
```
Comment (rafaelks) [inline Rocket.Chat/Controllers/Auth/AuthViewController.swift]: 👍

File: Rocket.Chat/Models/LoginService.swift [hunk 3]
```diff
@@ -43,6 +47,14 @@ class LoginService: BaseModel {
     @objc dynamic var loginStyle: String?
     @objc dynamic var buttonColor: String?
 
+    // CAS
+
+    @objc dynamic var loginUrl: String?
+
+    var isValid: Bool {
+        return !(serverUrl?.isEmpty ?? true) || type == .saml || type == .cas
```
Comment (rafaelks) [inline Rocket.Chat/Models/LoginService.swift]: I think this method deserves a comment... it validates if the connected server has any valid authentication method?

Comment (cardoso) [reply Rocket.Chat/Models/LoginService.swift]: It validates whether the LoginService has enough information to be used. I'll add a comment.

**Pull request merged**